SimplifiedTests/Mocks/NYPLOPDSFeedMock.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  NYPLOPDSFeedMock.swift
+//  Simplified
+//
+//  Created by Ernest Fan on 2022-01-04.
+//  Copyright © 2022 NYPL Labs. All rights reserved.
+//
+
+import Foundation
+@testable import SimplyE
+
+/// NYPLOPDSFeedMock is created in order to test NYPLCatalogUngroupedFeed,
+/// replacing the NYPLOPDSFeed class which fetches data from server and making the test results unreliable.
+class NYPLOPDSFeedMock: NYPLOPDSFeedProvider {
+  // A decremental counter which denies the next fetch once it reaches 0, mimicking an empty next URL
+  static var numberOfFetchAllowed = 15
+  
+  // Control the type of books in the returned feed
+  static var shouldReturnFeedWithSupportedBooks = true
+  
+  // Control for failing the fetch
+  static var shouldReturnFeed = true
+  
+  static func withURL(_ URL: URL!,
+                      shouldResetCache: Bool,
+                      completionHandler handler: ((NYPLOPDSFeed?, [AnyHashable : Any]?) -> Void)!) {
+    if !shouldReturnFeed {
+      handler(nil, nil)
+      return
+    }
+    
+    numberOfFetchAllowed -= 1
+    if numberOfFetchAllowed <= 0 {
+      shouldReturnFeed = false
+    }
+    
+    var filename:String = ""
+    
+    if (shouldReturnFeedWithSupportedBooks) {
+      filename = "NYPLCatalogUngroupedFeedWithSupportedBooks"
+    } else {
+      filename = "NYPLCatalogUngroupedFeedWithUnsupportedBooks"
+    }
+    
+    let feedUrl = Bundle.init(for: NYPLOPDSFeedMock.self)
```

Comment:
is this feedUrl ever different from what we receive in input (`URL`) ? It seems like we calculate the feedURL at the call sites (in `NYPLCatalogUngroupedFeedTests.swift`) the same way, so we could maybe just use URL here? I could be missing something.